docs/modules/clients/pages/java.adoc

```diff
@@ -862,14 +917,120 @@ Programmatic Configuration:
 ----
 ClientConfig clientConfig = new ClientConfig();
 ClientNetworkConfig networkConfig = clientConfig.getNetworkConfig();
-networkConfig.setSmartRouting(true);
+networkConfig.setClusterRoutingMode(ALL_MEMBERS);
 ----
 
-Its default value is `true` (smart client mode).
+**SINGLE_MEMBER**
+
+To connect to a single member, which can be used as a gateway to the other members, use the `SINGLE_MEMBER` cluster routing mode, which can be defined as described below.
 
-Note that you need to disable smart routing (`false`) for the clients which
-want to use temporary permissions defined in a member.
-See the xref:security:native-client-security.adoc#handling-permissions-when-a-new-member-joins[Handling Permissions section].
+When using the `SINGLE_MEMBER` cluster routing mode, consider the following:
+
+* The absence of <<configuring-backup-acknowledgement, backup acknowledgements>>, as the client does not have a view of the entire cluster
+* If you have multiple members on a single machine, we advise that <<configuring-address-list,explicit ports are set for each member>>
+* Connections to the xref:cp-subsystem:cp-subsystem.adoc[CP Subsystem] are direct-to-leader, which can result in improved performance
```

Comment:
This is only true if the client is explicitly set to connect to a CP group leader - and CP group leaders are defined algorithmically at runtime, so it's not possible to accurate predict which member will be a leader